src/XRP/serializer.ts

```diff
@@ -120,36 +106,23 @@ const serializer = {
     }
 
     // If an x-address was able to be decoded, add the components to the json.
-    const destinationAddress = payment.getDestination()
-    if (!destinationAddress) {
-      return undefined
-    }
-
-    const destination = destinationAddress.getValue()?.getAddress()
+    const destination = payment.getDestination()?.getValue()?.getAddress()
     if (!destination) {
       return undefined
     }
 
     const decodedXAddress = Utils.decodeXAddress(destination)
-    if (decodedXAddress) {
-      json.Destination = decodedXAddress.address
-      if (decodedXAddress.tag !== undefined) {
+    json.Destination = decodedXAddress?.address ?? destination
```

Comment:
Either it's an X-Address, or it was always a classic address (I think). Regardless of the validity of that assertion, this keeps the actual logic the same as before.